tutorials/stats-sensor-space/76_new_cluster_test_api.py

```diff
@@ -0,0 +1,491 @@
+):
+    """
+    Plot the cluster with the lowest p-value.
+
+    Parameters
+    ----------
+    contrast : list
+        List of contrast evoked objects.
+    target_only : list
+        List of target evoked objects.
+    non_target_only : list
+        List of non-target evoked objects.
+    T_obs : array
+        The observed test statistic.
+    clusters : list
+        List of clusters.
+    cluster_p_values : array
+        Array of cluster p-values.
+
+    Returns
+    -------
+    None
+
+    """
+    # configure variables for visualization
+    colors = {"target": "crimson", "non-target": "steelblue"}
+
+    # organize data for plotting
+    evokeds = {"target": target_only, "non-target": non_target_only}
+
+    lowest_p_cluster = np.argmin(cluster_p_values)
+
+    # plot the cluster with the lowest p-value
+    time_inds, space_inds = np.squeeze(clusters[lowest_p_cluster])
+    ch_inds = np.unique(space_inds)
+    time_inds = np.unique(time_inds)
+
+    # get topography for F stat
+    t_map = T_obs[time_inds, ...].mean(axis=0)
+
+    # get signals at the sensors contributing to the cluster
+    sig_times = contrast[0].times[time_inds]
+
+    # create spatial mask
+    mask = np.zeros((t_map.shape[0], 1), dtype=bool)
+    mask[ch_inds, :] = True
+
+    # initialize figure
+    fig, ax_topo = plt.subplots(1, 1, figsize=(10, 3), layout="constrained")
+
+    # plot average test statistic and mark significant sensors
+    t_evoked = mne.EvokedArray(t_map[:, np.newaxis], contrast[0].info, tmin=0)
+    t_evoked.plot_topomap(
+        times=0,
+        mask=mask,
+        axes=ax_topo,
+        cmap="Reds",
+        vlim=(np.min, np.max),
+        show=False,
+        colorbar=False,
+        mask_params=dict(markersize=10),
+    )
+    image = ax_topo.images[0]
+
+    # remove the title that would otherwise say "0.000 s"
+    ax_topo.set_title("")
+
+    # create additional axes (for ERF and colorbar)
+    divider = make_axes_locatable(ax_topo)
+
+    # add axes for colorbar
+    ax_colorbar = divider.append_axes("right", size="5%", pad=0.05)
+    plt.colorbar(image, cax=ax_colorbar)
+    ax_topo.set_xlabel(
+        "Averaged t-map ({:0.3f} - {:0.3f} s)".format(*sig_times[[0, -1]])
+    )
+
+    # add new axis for time courses and plot time courses
+    ax_signals = divider.append_axes("right", size="300%", pad=1.2)
+    title = f"Cluster #1, {len(ch_inds)} sensor"
+    if len(ch_inds) > 1:
+        title += "s (mean)"
+    mne.viz.plot_compare_evokeds(
+        evokeds,
+        title=title,
+        picks=ch_inds,
+        axes=ax_signals,
+        colors=colors,
+        show=False,
+        split_legend=True,
+        truncate_yaxis="auto",
+    )
+
+    # plot temporal cluster extent
+    ymin, ymax = ax_signals.get_ylim()
+    ax_signals.fill_betweenx(
+        (ymin, ymax), sig_times[0], sig_times[-1], color="green", alpha=0.3
+    )
+
+    plt.show()
+
+    return None
```

Comment:
this happens implicitly if no `return` is specified
```suggestion
```

Reply:
Okey, I thought it is always recommended to add a return statement?

Reply:
Nope. Only if you return something other than None, or if (inside a conditional branch) you want to return early